src/native/fs/fs_napi.cpp

```diff
@@ -1680,6 +1683,34 @@ struct FileFsync : public FSWrapWorker<FileWrap>
     }
 };
 
+struct FileFlock : public FSWrapWorker<FileWrap>
```

Comment:
would it make sense to add this as option to open instead?
do you need the UNLOCK option?

Reply:
I don't need `UNLOCK` yet. I separated it out from `open` so as to not make it seem like its a one step process (its not and that results in weird races).